src/hotspot/share/gc/x/xBarrier.inline.hpp

```diff
@@ -226,7 +226,7 @@ inline bool XBarrier::during_relocate() {
 // Load barrier
 //
 inline oop XBarrier::load_barrier_on_oop(oop o) {
-  return load_barrier_on_oop_field_preloaded((oop*)NULL, o);
+  return load_barrier_on_oop_field_preloaded((oop*)nullptr, o);
```

Comment:
Casts should not be needed on `nullptr`.

Reply:
They are needed to disambiguate the two overloaded functions that take `volatile oop*` and  `volatile narrowOop*` respectively.